pkg/cmd/experimental/sync-groups/interfaces.go

```diff
@@ -0,0 +1,32 @@
+package group
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/go-ldap/ldap"
+
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// LDAPGroupLister lists the LDAP groups that need to be synced by a job. The LDAPGroupLister needs to
+// be paired with an LDAPGroupDataExtractor that understands the format of the unique identifiers returned
+// to represent the LDAP groups to be synced.
+type LDAPGroupLister interface {
+	ListGroups() (groupUIDs []string, err error)
+}
+
+// LDAPGroupDataExtractor lists the members of an LDAP group as well as generating a GroupMeta for the LDAP group.
+type LDAPGroupDataExtractor interface {
```

Comment:
Add `LDAPGroupMapper` to this interface.  You've convinced me that they belong together.